plugins/ingest-attachment/src/main/java/org/elasticsearch/ingest/attachment/AttachmentProcessor.java

```diff
@@ -119,7 +119,12 @@ public void execute(IngestDocument ingestDocument) {
 
             if (properties.contains(Property.CONTENT_LENGTH)) {
                 String contentLength = metadata.get(Metadata.CONTENT_LENGTH);
-                String length = Strings.hasLength(contentLength) ? contentLength : String.valueOf(parsedContent.length());
+                long length;
+                if (Strings.hasLength(contentLength)) {
+                    length = Long.parseLong(contentLength);
```

Comment:
Indeed.

In the former mapper-attachments plugin we were having a setting to ignore metadata errors but continue with text extraction.
May be we should do that (in another PR though).